app.js

```diff
@@ -23,21 +23,19 @@ app.use(express.static(path.join(__dirname, 'public')));
 
 app.use('/', index);
 // catch 404 and forward to error handler
-app.use(function(req, res, next) {
-  var err = new Error('Not Found');
+app.use((req, res, next) => {
+  const err = new Error('Not Found');
   err.status = 404;
   next(err);
 });
 
-// error handler
-app.use(function(err, req, res, next) {
-  // set locals, only providing error in development
+app.use((err, req, res, next) => {
   res.locals.message = err.message;
   res.locals.error = req.app.get('env') === 'development' ? err : {};
-
   // render the error page
   res.status(err.status || 500);
   res.render('error');
+  next(err);
```

Comment:
It will break the server. when you have rendered the page, then no need to pass it to next middleware because its scope is ended.